src/modules/sync/utils/htmlRenderers.ts

```diff
@@ -0,0 +1,638 @@
+import {
+  ContentTypeElements,
+  ContentTypeModels,
+  ContentTypeSnippetModels,
+  TaxonomyModels,
+} from "@kontent-ai/management-sdk";
+
+import { LogOptions } from "../../../log.js";
+import {
+  isCountLimitation,
+  isDefaultValue,
+  isDependsOn,
+  isExternalIdReference,
+  isExternalIdReferenceArray,
+  isMaximumTextLength,
+  isObjectReference,
+  isObjectReferenceArray,
+  isValidationRegex,
+} from "../../../utils/typeguards.js";
+import { RequiredCodename } from "../../../utils/types.js";
+import { DiffModel, DiffObject } from "../types/diffModel.js";
+import {
+  AddIntoPatchOperation,
+  MovePatchOperation,
+  PatchOperation,
+  RemovePatchOperation,
+  ReplacePatchOperation,
+  ReplacePatchOperationValue,
+} from "../types/patchOperation.js";
+
+type EntityPathRenderer = {
+  regex: RegExp;
+  render: (() => string) | ((match: string[]) => string);
+};
+type Operation = PatchOperation["op"];
+type TypeOrSnippet = ContentTypeModels.IAddContentTypeData | ContentTypeSnippetModels.IAddContentTypeSnippetData;
+type ElementOrTerm = ContentTypeElements.Element | TaxonomyModels.IAddTaxonomyRequestModel;
+type RenderFunction<T extends PatchOperation> = (patchOp: T) => string;
+type EntityType = "taxonomies" | "types" | "snippets";
+type EntityActionType = "added" | "updated" | "deleted";
+type AdvancedDiffParams =
+  & Readonly<{
+    targetEnvironmentId: string;
+    outPath?: string;
+    sourceEnvironmentId?: string;
+    folderName?: string;
+    noOpen?: boolean;
+  }>
+  & LogOptions;
+
+export type DiffData = DiffModel & AdvancedDiffParams;
+
+export const resolveHtmlTemplate = (
```

Comment:
this sounds like a function to me that would have a great unit test :D it wouldn't be trivial because you would have to prepare some simple HTML to the string and create some example diffData object, but it doesn't have to be hard. What do you think?

Reply:
tl;dr

# `WON'T DO`

I'd say the function is already tested as a part of the integration test for advanced diff command, which basically just calls it. we can add a unit test later as well but I don't think it needs to be a part of this PR. I'm sure there will be some patch releases for things we may have missed, we can add the test at that point.